src/Book.ts

```
@@ -0,0 +1,197 @@
+            (result.credit - result.debit).toFixed(this.precision)
+          ),
+          notes: result.count,
+        };
+  }
+
+  async ledger<T = U>(
+    query: IParseQuery & IPaginationQuery,
+    populate?: string[] | null,
+    options?: IOptions & { lean?: true }
+  ): Promise<{ results: T[]; total: number }>;
+
+  async ledger<T = U>(
+    query: IParseQuery & IPaginationQuery,
+    populate?: string[] | null,
+    options?: IOptions & { lean?: false }
+  ): Promise<{ results: (Document & T)[]; total: number }>;
+
+  async ledger<T = U>(
+    query: IParseQuery & IPaginationQuery,
+    populate = null as string[] | null,
+    options = {} as IOptions & { lean?: boolean }
+  ): Promise<{ results: T[]; total: number }> {
+    let skip;
+    let limit = 0;
+
+    const { lean = true } = options;
+
+    // Pagination
+    if (query.perPage) {
+      skip = (query.page ? query.page - 1 : 0) * query.perPage;
+      limit = query.perPage;
+    }
+    const filterQuery = parseQuery(query, { name: this.name });
+    const q = transactionModel
+      .find(filterQuery, undefined, options)
+      .lean(lean)
+      .sort({
+        datetime: -1,
+        timestamp: -1,
+      });
+
+    let count = Promise.resolve(0);
+    if (typeof skip !== "undefined") {
+      count = transactionModel
+        .countDocuments(filterQuery)
+        .session(options.session || null)
+        .exec();
+      q.skip(skip).limit(limit);
+    }
+
+    if (populate) {
+      for (let i = 0, il = populate.length; i < il; i++) {
+        if (isValidTransactionKey<U>(populate[i])) {
+          q.populate(populate[i]);
+        }
+      }
+    }
+
+    const results = (await q.exec()) as unknown as T[];
+
+    return {
+      results,
+      total: (await count) || results.length,
+    };
+  }
+
+  async void(
+    journal_id: string,
+    reason?: undefined | string,
+    options = {} as IOptions
+  ) {
+    const journal: TJournalDocument<J> = (await journalModel
```

Maybe we should throw an Error if we dont find the requested Journal?

---

Now it is book-scoped and throws an error if we dont find thejournal within the book